anthos-bm-openstack-terraform/docs/quickstart.md

```diff
@@ -0,0 +1,512 @@
+
+#### 3.1) Find the public `network id` in your OpenStack environment
+```sh
+# you can use this one line command if you have `jq` CLI tool installed
+# if not use the "openstack network list --name=public" command to get the ID
+export PUBLIC_NETWORK_ID=$(openstack network list --name=public -f json | jq -c '.[]."ID"' | tr -d '"')
+
+# make sure you have the PUBLIC_NETWORK_ID environment variable is set
+echo $PUBLIC_NETWORK_ID
+```
+
+#### 3.2) Generate the Terraform variables file
+```sh
+cat > terraform.tfvars << EOF
+external_network_id = "$PUBLIC_NETWORK_ID"
+os_user_name        = "$OS_USERNAME"
+os_tenant_name      = "$OS_TENANT_NAME"
+os_password         = "$OS_PASSWORD"
+os_auth_url         = "$OS_AUTH_URL"
+os_endpoint_type    = "$OS_ENDPOINT_TYPE"
+ssh_key_name        = "$SSH_KEY_NAME"
+EOF
+
+# see it's contents
+cat terraform.tfvars
+```
+```sh
+# -----------------------------------------------------
+#                   Expected Output
+# -----------------------------------------------------
+external_network_id = "abce7af2-db6a-47db-b4dc-04598a73ec2d"
+os_user_name        = "admin"
+os_tenant_name      = ""
+os_password         = "54e53c4ecc76fa84bec1374894583b8651332e0abe45c1de421ba797f669f35"
+os_auth_url         = "https://10.128.0.2:5000"
+os_endpoint_type    = ""
+ssh_key_name        = "abmNodeKey"
+```
+
+#### 3.3) Initialize Terraform
+```sh
+# this sets up the required Terraform state management configurations, similar to 'git init'
+terraform init
+```
+
+#### 3.4) Create a _Terraform execution_ plan
+```sh
+# compares the state of the resources, verifies the scripts and creates an execution plan
+terraform plan
+```
+
+#### 3.5) Apply the changes described in the Terraform script
+Review the **Terraform** script _([main.tf](/main.tf))_ to see details of the
+configuration. Running this script will create the required VMs and setup the
+networking inside **OpenStack** to install **Anthos on Bare Metal**.
+
+```sh
+# executes the plan on the given provider (i.e: GCP) to reach the desired state of resources
+terraform apply
+```
+```sh
+# -----------------------------------------------------
+#                   Expected Output
+# -----------------------------------------------------
+Apply complete! Resources: 18 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+admin_ws_public_ip = "172.29.249.165"
+```
+> **Note 1:** When prompted to confirm the Terraform plan, type 'Yes' and enter
+>
+> **Note 2:** This step can take upto **90 seconds** to complete
+
+---
+### 4. Configure the admin workstation VM on OpenStack
+Running the **Terraform** scripts from section [3.5](#35-apply-the-changes-described-in-the-terraform-script)
+would have created a VM in **OpenStack** that will serve as our ***admin workstation***.
+We will confgure and use it to install **Anthos on Bare Metal**.
+
+#### 4.1) Fetch the Floating IP of the admin workstation
+> **Note:** Use one of the two commands below to setup the environment variable
```

Comment:
The dual option is because the devsite document assumes that the users already have an OpenStack environment running and states that they can either do one of the 2 before attempting to install ABM:
- Manually create the VMs and networking in OpenStack looking at the diagram
- Use the terraform scripts to create the VMs and networking to match the diagram

So I want to cater for both cases of followers arriving at this step. Ideally the `openstack` based command should work for both. But, thought it might be useful to share the terraform one as well.